### PR TITLE
create get problems API

### PR DESCRIPTION
**Pull request by cameroncuster**
- added the API for getting the entire problems repository with Octokit (subject to rate limiting)

### CODE REVIEW

File: functions/src/index.ts [hunk 1]
```diff
@@ -97,4 +98,13 @@ app.post('/registerWithEmail', emailRegister)
  */
 app.post('/getUserData', getUserData)
 
+/**
+ * API for retrieving problems
+ *
+ * @req Empty
+ * @res Returns an array of Problem objects
```
Comment (AsirAAlam) [inline functions/src/index.ts:105]: nit: i'd probably specify that it returns ALL problems

File: package.json [hunk 2]
```diff
@@ -1,10 +1,12 @@
 {
   "dependencies": {
+    "@octokit/rest": "^20.0.2",
     "cors": "^2.8.5",
     "express": "^4.18.2",
     "firebase": "^10.5.2",
     "firebase-admin": "^11.11.0",
     "firebase-functions": "^4.5.0",
+    "octokit": "^3.1.2",
```
Comment (AsirAAlam) [inline package.json:9]: is this import necessary? is the @octokit/rest above not sufficient?

**Review by AsirAAlam — APPROVED** [review]
Looks great! Small comments.

Also, can you include your testing process?

File: package.json (hunk 2 above)
Comment (cameroncuster) [reply package.json:9]: yeah I added both when I was trying to get it to work, I'll play around with it

**Comment by gblake31** [comment]
Let's see if we can get the authentication part working before we merge this.

**Comment by cameroncuster** [comment]
Closing this because we decided to move forward with a new approach

**Pull request closed without merging**